src/Workspaces/Core/Portable/Extensions/OperationExtensions.cs

```diff
@@ -43,51 +53,78 @@ public static ValueUsageInfo GetValueUsageInfo(this IOperation operation)
+                        return operation.Parent.Kind == OperationKind.CompoundAssignment
+                            ? ValueUsageInfo.ReadWrite
+                            : ValueUsageInfo.Write;
+                    }
 
-                    case RefKind.Out:
-                        return ValueUsageInfo.WritableRef;
+                    break;
 
-                    case RefKind.Ref:
-                        return ValueUsageInfo.ReadableWritableRef;
+                case IIncrementOrDecrementOperation _:
+                    return ValueUsageInfo.ReadWrite;
 
-                    default:
-                        return ValueUsageInfo.Read;
-                }
-            }
-            else if (IsInLeftOfDeconstructionAssignment(operation))
-            {
-                return ValueUsageInfo.Write;
+                case IInvalidOperation invalidOperation:
+                    isInErrorContext = true;
+                    if (invalidOperation.IsImplicit)
+                    {
+                        return invalidOperation.GetValueUsageInfo(out _);
+                    }
+
+                    break;
+
+                case IConversionOperation conversionOperation:
+                    var result = conversionOperation.GetValueUsageInfo(out var isParentInErrorContext) | ValueUsageInfo.Read;
+                    isInErrorContext |= isParentInErrorContext;
+                    return result;
+
+                case IParenthesizedOperation parenthesizedOperation:
+                    // Note: IParenthesizedOperation is specific to VB, where the parens cause a copy, so this cannot be classified as a write.
+                    Debug.Assert(parenthesizedOperation.Language == LanguageNames.VisualBasic);
+
+                    return parenthesizedOperation.GetValueUsageInfo(out isInErrorContext) &
+                        ~(ValueUsageInfo.Write | ValueUsageInfo.WritableRef);
+
+                case INameOfOperation _:
+                case ITypeOfOperation _:
+                case ISizeOfOperation _:
+                    return ValueUsageInfo.NonReadWriteRef;
+
+                case IArgumentOperation argumentOperation:
+                    switch (argumentOperation.Parameter.RefKind)
+                    {
+                        case RefKind.RefReadOnly:
+                            return ValueUsageInfo.ReadableRef;
+
+                        case RefKind.Out:
+                            return ValueUsageInfo.WritableRef;
+
+                        case RefKind.Ref:
+                            return ValueUsageInfo.ReadableWritableRef;
+
+                        default:
+                            return ValueUsageInfo.Read;
+                    }
+
+                default:
+                    if (IsInLeftOfDeconstructionAssignment(operation))
+                    {
+                        return ValueUsageInfo.Write;
+                    }
+
+                    break;
             }
 
             return ValueUsageInfo.Read;
```

Comment:
is this the right default fall-out case?  I'd almost prefer an assert here so we catch this and have to explicitly define all the cases.